### ISSUE TITLE
 '_SH_DENYWR' was not declared in this scope

### ISSUE THREAD

**Issue by DJuego** [gabime/spdlog#84]
Any compiler errors with mingw-builds (i686-4.9.2-release-win32-sjlj-rt_v4-rev2) [WIN32]

..\..\inc/spdlog/sinks/../details/./os.h: In function 'int spdlog::details::os::fopen_s(FILE**, const string&, const char*)':
..\..\inc/spdlog/sinks/../details/./os.h:150:45: error: '_SH_DENYWR' was not declared in this scope
     *fp = _fsopen((filename.c_str()), mode, _SH_DENYWR);

I can solve it with:

File os.h:
```cpp
#ifdef _WIN32
#ifndef WIN32_LEAN_AND_MEAN
#define WIN32_LEAN_AND_MEAN
#endif
#include <Windows.h>
+#include <share.h>
```

**Comment by gabime** [comment]
What windows version do you use?

**Comment by gabime** [comment]
I committed a fix   - 80fcd655649eca5fec85c70b7016c6a63baa5100
